Dieudonne complete (#1426)
pi-base/data: properties/P000221.md

---
uid: P000221
name: Dieudonné complete
aliases:
  - Completely uniformizable
  - Topologically complete
refs:
  - wikipedia: Uniform_space
    name: Uniform space on Wikipedia
  - wikipedia: Completely_uniformizable_space
    name: Completely uniformizable space
  - doi: 10.1007/978-1-4615-7819-2
    name: Rings of Continuous Functions (Gillman & Jerison)
  - mr: 370454
    name: General Topology (Kelley)
  - zb: "0684.54001"
    name: General Topology (Engelking, 1989)
---

The topology on the space $X$ is induced by a [complete uniformity](https://en.wikipedia.org/wiki/Uniform_space#Completeness) $\mathcal{U}$.

> We call uniformity $\mathcal{U}$ complete if every Cauchy filter $\mathcal{F}$ on $(X, \mathcal{U})$ converges. Equivalently, every Cauchy net $(x_i)_{i\in I}$ on $(X, \mathcal{U})$ converges. Here Cauchy filter is a filter $\mathcal{F}$ such that for every $U\in\mathcal{U}$ there exists $A\in\mathcal{F}$ such that $A\times A\subseteq U$. A Cauchy net is a net $(x_i)_{i\in I}$ such that for every $U\in\mathcal{U}$ there exists $i_0$ such that $(x_j, x_k)\in U$ for $j, k\geq i_0$.

Reviewer: ```suggestion
We call a uniformity $\mathcal{U}$ *complete* if every Cauchy filter $\mathcal{F}$ on $(X, \mathcal{U})$ converges. Equivalently, every Cauchy net $(x_i)_{i\in I}$ on $(X, \mathcal{U})$ converges. Here a *Cauchy filter* is a filter $\mathcal{F}$ such that for every $U\in\mathcal{U}$ there exists $A\in\mathcal{F}$ such that $A\times A\subseteq U$. A *Cauchy net* is a net $(x_i)_{i\in I}$ such that for every $U\in\mathcal{U}$ there exists $i_0$ such that $(x_j, x_k)\in U$ for $j, k\geq i_0$.
```

If $X$ is $T_0$ this is equivalent to the following (see {{zb:0684.54001}} exercise 8.5.13a):

1. $X$ is a closed subspace of a product of {P55} spaces
2. $X$ is a closed subspace of a product of {P53} spaces.

Compare with definition in 15.7 of {{doi:10.1007/978-1-4615-7819-2}} where uniform structure is defined using pseudometrics instead.

> This property is *topologically complete* in {{mr:370454}}.

Reviewer: ```suggestion
This property is usually called *Dieudonné complete* (Problem 8.5.13 in {{zb:0684.54001}}) or *completely uniformizable* (Problem 39B in {{zb:1052.54001}}).
```
and now need to add Willard to the `refs:`

---
uid: P000221
name: Dieudonné complete
aliases:
  - Completely uniformizable
  - Topologically complete
refs:
  - wikipedia: Uniform_space
    name: Uniform space on Wikipedia
  - wikipedia: Completely_uniformizable_space
    name: Completely uniformizable space
  - doi: 10.1007/978-1-4615-7819-2
    name: Rings of Continuous Functions (Gillman & Jerison)
  - mr: 370454
    name: General Topology (Kelley)
  - zb: "0684.54001"
    name: General Topology (Engelking, 1989)
  - zb: "1052.54001"
    name: General Topology (Willard, 1970)
---

The topology on the space $X$ is induced by a [complete uniformity](https://en.wikipedia.org/wiki/Uniform_space#Completeness) $\mathcal{U}$.

We call a uniformity $\mathcal{U}$ *complete* if every Cauchy filter $\mathcal{F}$ on $(X, \mathcal{U})$ converges. Equivalently, every Cauchy net $(x_i)_{i\in I}$ on $(X, \mathcal{U})$ converges. Here a *Cauchy filter* is a filter $\mathcal{F}$ such that for every $U\in\mathcal{U}$ there exists $A\in\mathcal{F}$ such that $A\times A\subseteq U$. A *Cauchy net* is a net $(x_i)_{i\in I}$ such that for every $U\in\mathcal{U}$ there exists $i_0$ such that $(x_j, x_k)\in U$ for $j, k\geq i_0$.

> If $X$ is $T_0$ this is equivalent to the following (see {{zb:0684.54001}} exercise 8.5.13a):

Reviewer: ```suggestion
If $X$ is $T_0$ this is equivalent to the following
(see Problem 8.5.13(a) in {{zb:0684.54001}}, which assumes $T_1$ in the definition of uniform space):
```
this is for discussion.  We can mention something else than T1, but I think it's worth mentioning that Engelking is assuming something extra).

Reviewer: that's why I was wondering if maybe it's better to adjust things to assume T1 instead of T0 at the beginning (it's really all the same as Tychonoff anyway).  Or use Hausdorff?

Reviewer: or maybe say "..., which assumes extra separation axioms in the definition of uniform space" ? (or of uniformity)

Author: @prabau No, I don't like this change

> If $X$ is $T_0$ this is equivalent to the following (see {{zb:0684.54001}} exercise 8.5.13a):

Reviewer 2: Instead of requiring t0, couldnt we require that
1. Kol(X)  is closed subspace of P55 spaces
2. X is closed subspace of P121 spaces

?
Then this could be merged with the section before, i.e.

"The topology of X is induced by a complete uniformity, meaning ...

Equivalently:

- every cauchy filter converges
- Kol(X)  is closed subspace of P55 spaces
- X is closed subspace of P121 spaces

Author: @felixpernegger we could, but then the statement would be slightly more unnatural, and also wouldn't be faithful to exposition in Engelking

Author: I also feel like telling things in terms of pseudometrizable spaces feels a bit "odd"

Author: I've been thinking about it again, and I feel like maybe it is a good idea after all. But maybe not in the exact same phrasing like you did here, I am think "product of pseudometrizable" and "product of complete pseudometrizable" spaces

1. $X$ is a closed subspace of a product of {P55} spaces
2. $X$ is a closed subspace of a product of {P53} spaces.

Compare with definition in 15.7 of {{doi:10.1007/978-1-4615-7819-2}} where uniform structure is defined using pseudometrics instead.

> This property is usually called *Dieudonné complete* (Problem 8.5.13 in {{zb:0684.54001}}) or *completely uniformizable* (Problem 39B in {{zb:1052.54001}}). It is called *topologically complete* on page 208 of {{mr:370454}}

Reviewer: ```suggestion
This property is usually called *Dieudonné complete* (Problem 8.5.13 in {{zb:0684.54001}}) or *completely uniformizable* (Problem 39B in {{zb:1052.54001}}).
It is called *topologically complete* on page 208 of {{mr:370454}}
(but the same term has also been used for {P55} and for {P63}).
```

Author: @prabau wouldn't we need to add the same statements for both P55 and P63 then? Isn't it better to not add it?

Reviewer: My sources: nlab and Encycl. of Gen Top.  Also searched to "topologically complete" in google scholar and saw there were many hits, but I did not pursue further.

P55 already has "topologically complete" as an alias.  P63 doesn't, but if it's also used in that sense, I think we can add it and we should look for a more specific sample source.  And it would make sense to add a corresponding sentence linking to the other two in each case.

At least that would reflect what has been used out there.

Author: @prabau well, it'll be slightly annoying, but sure, I'll add it 😓

Author: @prabau do you have an idea where I can find source for Cech complete being called topologically complete

Edit: I have found an article like that by well-known mathematician, but maybe a book would be better

Author: @prabau I have added this in some form but decided against your phrasing. I feel like it's incorrect to say that the name "topologically complete" is somehow different from the others, and so we should get rid of the biased language.

    name: Uniform space on Wikipedia
  - wikipedia: Completely_uniformizable_space
    name: Completely uniformizable space
  - doi: 10.1007/978-1-4615-7819-2
    name: Rings of Continuous Functions (Gillman & Jerison)
  - mr: 370454
    name: General Topology (Kelley)
  - zb: "0684.54001"
    name: General Topology (Engelking, 1989)
  - zb: "1052.54001"
    name: General Topology (Willard, 1970)
---

The topology on the space $X$ is induced by a [complete uniformity](https://en.wikipedia.org/wiki/Uniform_space#Completeness). 

We call a uniformity $\mathcal{U}$ *complete* if every Cauchy filter $\mathcal{F}$ on $(X, \mathcal{U})$ converges. Equivalently, every Cauchy net $(x_i)_{i\in I}$ on $(X, \mathcal{U})$ converges. Here a *Cauchy filter* is a filter $\mathcal{F}$ such that for every $U\in\mathcal{U}$ there exists $A\in\mathcal{F}$ such that $A\times A\subseteq U$. A *Cauchy net* is a net $(x_i)_{i\in I}$ such that for every $U\in\mathcal{U}$ there exists $i_0$ such that $(x_j, x_k)\in U$ for $j, k\geq i_0$.

The following are equivalent (compare to exercise 8.5.13a of {{zb:0684.54001}}):

1. $X$ is {P221},
2. $X$ is a closed subspace of a product of completely {P121} spaces,

> 3. $X$ is a closed subspace of a product of {P121} spaces.

Reviewer 2: Thinking about it, it is probably fine now.

I dont like this phrasing so much (as it puts arguably the simpler definition to what it feels like a lower rank), but at the same time i cant think of a better solution.

Author: @felixpernegger what do you mean by simpler definition? The ones about products?

Reviewer 2: yes

Reviewer 2: But at the same time "Completely uniformizable" of course morally means "induced by complete uniformity", which is why it should stay as first deifnition

Author: Yep. That's my thoughts too. I'm also reminded of @StevenClontz and @prabau trying to see if we can define realcompactness in terms of closed subspace of product of real lines. I had to convince them that this is only a corollary of the "actual" definition of realcompactness.

Reviewer: I agree "induced by a complete uniformity" should be the primary one.

Compare this definition of complete uniformity with the definition in 15.7 of {{zb:1380.46022}} where uniform structure is defined using pseudometrics instead. 

> Such spaces are called *Dieudonné complete* (Problem 8.5.13 in {{zb:0684.54001}}), *completely uniformizable* (Problem 39B in {{zb:1052.54001}}) or *topologically complete* (Page 208 of {{mr:370454}}). The last term is also used for {P55} and {P63}.

Reviewer: ```suggestion
Such spaces are called *Dieudonné complete* (Problem 8.5.13 in {{zb:0684.54001}}) or *completely uniformizable* (Problem 39B in {{zb:1052.54001}}).
They are called *topologically complete* on Page 208 of {{mr:370454}}; 
this last term has also been used for {P55} and {P63}.
```

Reviewer: I'll incorporate this into a bigger suggestion for this file.

---
uid: P000221
name: Dieudonné complete
aliases:
  - Completely uniformizable
  - Topologically complete
refs:
  - wikipedia: Completely_uniformizable_space
    name: Completely uniformizable space
  - zb: "1380.46022"
    name: Rings of Continuous Functions (Gillman & Jerison)
  - mr: 370454
    name: General Topology (Kelley)
  - zb: "0684.54001"
    name: General Topology (Engelking, 1989)
  - zb: "1052.54001"
    name: General Topology (Willard, 1970)
  - zb: "0024.36301"
    name: Sur les espaces uniformes complets (Dieudonné)
---

> The topology on $X$ is induced by a [complete uniformity](https://en.wikipedia.org/wiki/Uniform_space#Completeness).

Reviewer: ```suggestion
There exists at least one [complete uniformity](https://en.wikipedia.org/wiki/Uniform_space#Completeness)
that induces the topology of $X$.
```
One could think that if one uniformity compatible with the topology is complete, all of them are.
I think the suggestion here better hints that this is not the case.
